test-adapter/src/main/scala/org/scalajs/sbttestadapter/ComJSEnvRPC.scala

```diff
@@ -0,0 +1,67 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js sbt plugin        **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013, LAMP/EPFL        **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+
+
+package org.scalajs.testadapter
+
+import scala.concurrent._
+import scala.concurrent.duration._
+
+import org.scalajs.jsenv.{ComJSEnv, ComJSRunner}
+import org.scalajs.testcommon._
+
+/** RPC Core for use with a [[ComJSRunner]].
+ *
+ *  @note You may only create one instance of this per [[ComJSRunner]]
+ */
+private[testadapter] final class ComJSEnvRPC(
+    val runner: ComJSRunner) extends RPCCore {
+
+  private[this] var closed = false
+
+  private val receiverThread = new Thread {
+    setName("ComJSEnvRPC receiver")
+
+    override def run(): Unit = {
+      try {
+        while (true) {
+          try {
+            handleMessage(runner.receive())
+          } catch {
+            case _: TimeoutException =>
+          }
+        }
+      } catch {
+        case _: ComJSEnv.ComClosedException       =>
+        case _: InterruptedException              =>
+        case _: Exception if synchronized(closed) =>
```

Comment:
Good catch. Needed to make `closed` volatile, because I actually can't synchronize here :) Another thread holding that monitor is waiting for this thread to terminate :)